### ISSUE TITLE
Error in running Leiden algorithm

### ISSUE THREAD

**Issue by Jay-Leung** [crazyhottommy/pyflow_seuratv3_parameter#13]
Hi Ming Tang,

I previously ran pyflow_seurat_parameter with the default Louvain algorithm and it worked. However, when I changed the algorithm to Leiden (i.e. algorithm = 4) after installing leidenalg, the process kept getting killed during the run. I am not sure what went wrong, the log files do not show any specific error. I have attached the terminal output for your reference. Could you take a look and see what may be the issue?

[Terminal Output.txt](https://github.com/crazyhottommy/pyflow_seurat_parameter/files/7266782/Terminal.Output.txt)

**Comment by Jay-Leung** [comment]
[full_sample_k_20_resolution_0.6_PC_40.log](https://github.com/crazyhottommy/pyflow_seurat_parameter/files/7366197/full_sample_k_20_resolution_0.6_PC_40.log)
[full_sample_k_20_resolution_0.8_PC_40.log](https://github.com/crazyhottommy/pyflow_seurat_parameter/files/7366198/full_sample_k_20_resolution_0.8_PC_40.log)
[full_sample_k_100_resolution_1_PC_20.log](https://github.com/crazyhottommy/pyflow_seurat_parameter/files/7366199/full_sample_k_100_resolution_1_PC_20.log)

Hi, I realised I should have uploaded the log files as well, since it is mentioned in the terminal output. As you can see, there is no error reported. Could you perhaps try running the leiden algorithm on your end and see if it works?

Thanks!

**Comment by crazyhottommy** [comment]
Hi, if you just run the R command in the terminal and specify the algorithm =4, did it go through? that's how I usually debug.

**Comment by Jay-Leung** [comment]
Hi Ming Tang, 

I ran it in the terminal and it went through, but when I ran with snakemake it failed again. I realised this is a memory issue - somehow running Leiden uses more memory than Louvain, so I had to reduce the core number. Thank you so much for your suggestion!

**Comment by crazyhottommy** [comment]
great!

**Comment by Jay-Leung** [comment]
Hi @crazyhottommy,

Just a question out of curiosity. Previously I installed leidenalg in the python snakemake venv itself on a local workstation and ran pyflow_seurat_parameter with 3 cores, which threw an error (due to memory consumption but I wasn't aware at that time). I then posted this issue and tried what you suggested by running the command in the R terminal, which prompted me to install leidenalg through r-reticulate venv. Eventually by decreasing the cores to 2, it was able to run smoothly. 

However, when I deleted the r-reticulate venv (since it made sense that the script was using leidenalg in the snakemake venv), I realised that it could no longer run on 2 cores, again due to memory consumption (it was hitting above 128 GB on 2 cores, which is the max memory of my workstation). So I had to lower it to 1 core to run the workflow. My question is - why are there memory differences when interfacing leidenalg from r-reticulate vs snakemake venv? Is it to do with how the script is written to interface the specified venv?

Thanks so much!

Regards,
Jay
